### PR TITLE
Add egl capabilities to compositor

### PR DESCRIPTION
**Pull request by neithernut**
This PR aims for integration of EGL capablities to the compositor.
Targets #15.

I'll probably discuss a lot of stuff (e.g. with @TheNeikos) while implementing this, so this PR may progress quite slowly.

### CODE REVIEW

**Comment by neithernut** [comment]
Question: should `ws_framebuffer_device` have a function for creating a buffer on top of a `gbm_surface`?

**Comment by matthiasbeyer** [comment]
I'll wait with reviewing until the WIP tag is removed.

**Comment by TheNeikos** [comment]
> I'll probably discuss a lot of stuff (e.g. with @TheNeikos) while implementing this, so this PR may progress quite slowly.

Implying I am slow :c

**Comment by TheNeikos** [comment]
Added libgbm to the image

**Comment by neithernut** [comment]
_Almost_ done here. The only thing left to do is figuring out what attributes we need.

File: src/compositor/buffer/egl.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+ * waysome is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with waysome. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// malloc _mus_ be included before gbm.h
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl.c]: typo.

File: src/compositor/buffer/egl.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        .supertype  = (ws_object_type_id*)&WS_OBJECT_TYPE_ID_BUFFER,
+        .typestr    = "ws_egl_buffer",
+
+        .hash_callback = NULL,
+
+        .deinit_callback = NULL,
+        .cmp_callback = NULL,
+
+        .attribute_table = NULL,
+    },
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl.c:68]: Missing:
- `uuid_callback`
- `function_table`

File: src/compositor/buffer/egl.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    EGLint const attrs[] = {
+        //!< @todo fill in the real attributes
+        EGL_NONE
+    };
+
+    EGLConfig egl_conf = NULL;
+    EGLint conf_num = 0;
+    if (!eglChooseConfig(egl_disp, attrs, &egl_conf, 1, &conf_num) ||
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl.c]: It would be nice to have a short note what the hardcoded `1` is for.

File: src/compositor/buffer/egl.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+};
+
+/**
+ * Variable which holds the type information about the ws_egl_buffer type
+ */
+extern ws_buffer_type_id WS_OBJECT_TYPE_ID_EGL_BUFFER;
+
+/**
+ * Create a new EGL buffer from a gbm device, an EGL display and config
+ */
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl.h:66]: Please add a `@memberof` and maybe a `@return` (I know this is obvious, ...).

File: src/compositor/buffer/egl.c (hunk 3 above)
Comment (neithernut) [inline src/compositor/buffer/egl.c]: If you read the documentation of `eglChooseConfig()`, it's obvious. An extra comment would result in an extremly ugly block.

Comment (matthiasbeyer) [inline src/compositor/buffer/egl.c]: Ok.

**Comment by neithernut** [comment]
I decided to take the EGL attributes from the caller, since I have no idea what the caller may want (well, for internal purposes mybe, but not for clients).
So far, I'm done.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**